pkg/descheduler/strategies/lownodeutilization.go

```diff
@@ -269,7 +288,30 @@ func SortNodesByUsage(nodes []NodeUsageMap) {
 	})
 }
 
-func CreateNodePodsMap(client clientset.Interface, nodes []*v1.Node) NodePodsMap {
+// sortPodsBasedOnPriority sorts pods based on priority and if their priorities are equal, they are sorted based on QoS tiers.
+func sortPodsBasedOnPriority(evictablePods []*v1.Pod) {
+	sort.Slice(evictablePods, func(i, j int) bool {
+		if evictablePods[i].Spec.Priority == nil {
+			*evictablePods[i].Spec.Priority = 0 // This should have been handled by priority admission controller but to be on safe-side, we are setting its priority to 0.
+		}
+		if evictablePods[j].Spec.Priority == nil {
+			*evictablePods[j].Spec.Priority = 0 // This should have been handled by priority admission controller but to be on safe-side, we are setting its priority to 0.
+		}
+		if *evictablePods[i].Spec.Priority == *evictablePods[j].Spec.Priority {
+			if podutil.IsBestEffortPod(evictablePods[i]) && (podutil.IsGuaranteedPod(evictablePods[j]) || podutil.IsBurstablePod(evictablePods[j])) {
+				return true
+			}
+			if podutil.IsBurstablePod(evictablePods[i]) && podutil.IsGuaranteedPod(evictablePods[j]) {
+				return true
```

Comment:
you have not checked for the scenario when `j` is best effort, and `i` is burstable.

Reply:
Nevermind, it is covered by false below.

Reply:
Yes.